src/main/java/com/blackduck/integration/jenkins/polaris/service/PolarisCliVersionHandler.java

```diff
@@ -0,0 +1,55 @@
+package com.blackduck.integration.jenkins.polaris.service;
+
+import com.fasterxml.jackson.core.Version;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PolarisCliVersionHandler {
+    private static final String POLARIS_VERSION_PATTERN =
+            "Polaris Software Integrity Platform CLI Scan Client version - (\\d+\\.\\d+\\.\\d+)";
+
+    public String extractPolarisCliVersion(InputStream logStream) throws IOException {
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(logStream, StandardCharsets.UTF_8))) {
+            String line;
+            Pattern pattern = Pattern.compile(POLARIS_VERSION_PATTERN);
+            while ((line = reader.readLine()) != null) {
+                Matcher matcher = pattern.matcher(line);
+                if (matcher.find()) {
+                    return matcher.group(1);
+                }
+            }
+        }
+        return null;
+    }
+
+    public int comparePolarisVersions(String version1, String version2) {
```

Comment:
How about adding comment for this function? This will specially help anyone understand the return value.

Reply:
Added detailed comment